models/models.go

```diff
@@ -1,17 +1,20 @@
 package models
 
 type VampConfig struct {
-	RootPassword          string `yaml:"rootPassword,omitempty" json:"rootPassword,omitempty"`
-	DatabaseUrl           string `yaml:"databaseUrl,omitempty" json:"databaseUrl,omitempty"`
-	DatabaseName          string `yaml:"databaseName,omitempty" json:"databaseName,omitempty"`
-	ImageName             string `yaml:"imageName,omitempty" json:"imageName,omitempty"`
-	RepoUsername          string `yaml:"repoUsername,omitempty" json:"repoUsername,omitempty"`
-	RepoPassword          string `yaml:"repoPassword,omitempty" json:"repoPassword,omitempty"`
-	ImageTag              string `yaml:"imageTag,omitempty" json:"imageTag,omitempty"`
-	Mode                  string `yaml:"mode,omitempty" json:"mode,omitempty"`
-	AccessTokenExpiration string `yaml:"accessTokenExpiration,omitempty" json:"accessTokenExpiration,omitempty"`
-	IstioInstallerImage   string `yaml:"istioInstallerImage,omitempty" json:"istioInstallerImage,omitempty"`
-	IstioAdapterImage     string `yaml:"istioAdapterImage,omitempty" json:"istioAdapterImage,omitempty"`
+	RootPassword                   string `yaml:"rootPassword,omitempty" json:"rootPassword,omitempty"`
+	DatabaseUrl                    string `yaml:"databaseUrl,omitempty" json:"databaseUrl,omitempty"`
+	DatabaseName                   string `yaml:"databaseName,omitempty" json:"databaseName,omitempty"`
+	ImageName                      string `yaml:"imageName,omitempty" json:"imageName,omitempty"`
+	RepoUsername                   string `yaml:"repoUsername,omitempty" json:"repoUsername,omitempty"`
+	RepoPassword                   string `yaml:"repoPassword,omitempty" json:"repoPassword,omitempty"`
+	ImageTag                       string `yaml:"imageTag,omitempty" json:"imageTag,omitempty"`
+	Mode                           string `yaml:"mode,omitempty" json:"mode,omitempty"`
+	AccessTokenExpiration          string `yaml:"accessTokenExpiration,omitempty" json:"accessTokenExpiration,omitempty"`
+	IstioInstallerImage            string `yaml:"istioInstallerImage,omitempty" json:"istioInstallerImage,omitempty"`
+	IstioAdapterImage              string `yaml:"istioAdapterImage,omitempty" json:"istioAdapterImage,omitempty"`
+	MinReplicas                    *int32 `yaml:"minReplicas,omitempty" json:"minReplicas,omitempty"`
```

Comment:
why are these optional and Max replicas is not?
And why are not validated?
We can make all of them optional and set default values if there are nil and min replicas should always be less than 0. Target CPU utilization should be between 0 and 100 as I remember.

Reply:
MinReplicas and TargetCPUUtilization can be skipped - MinReplicas has default 1 and if there is no TargetCPUUtilization than the default autoscaling policy is used. 
I'll add validation.

Reply:
ok.